### PR TITLE
ARC-396: add lambda cron example for terraform

### PR DESCRIPTION
**Pull request by shubhamsinha-sf**
- add a lambda triggered by an eventbridge cron event for terraform

### CODE REVIEW

**Review by mayank0202 — APPROVED** [review]

**Review by tsaucier-sf — APPROVED** [review]

File: aws-lambdas/iac/terraform/lambda/main.tf [hunk 1]
```diff
@@ -1,3 +1,17 @@
+terraform {
+  required_version = ">= 1.0.8"
```
Comment (tsaucier-sf) [inline aws-lambdas/iac/terraform/lambda/main.tf:2]: Don't make this a hard constraint. Change to `required_version = "~> 1.3"`

File: aws-lambdas/iac/terraform/lambda/main.tf [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "4.20.1"
```
Comment (tsaucier-sf) [inline aws-lambdas/iac/terraform/lambda/main.tf:6]: Change this to `version = "~> 4.20"`

File: aws-lambdas/iac/terraform/lambda/main.tf [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+    }
+    archive = {
+      source  = "hashicorp/archive"
+      version = "2.2.0"
```
Comment (tsaucier-sf) [inline aws-lambdas/iac/terraform/lambda/main.tf:10]: Change this to `version = "~> 2.2"`

**Review by tsaucier-sf — CHANGES_REQUESTED** [review]

File: aws-lambdas/iac/terraform/lambda/main.tf (hunk 1 above)
Comment (shubhamsinha-sf) [reply aws-lambdas/iac/terraform/lambda/main.tf]: done

File: aws-lambdas/iac/terraform/lambda/main.tf (hunk 2 above)
Comment (shubhamsinha-sf) [reply aws-lambdas/iac/terraform/lambda/main.tf]: done

File: aws-lambdas/iac/terraform/lambda/main.tf (hunk 3 above)
Comment (shubhamsinha-sf) [reply aws-lambdas/iac/terraform/lambda/main.tf]: done

**Review by tsaucier-sf — APPROVED** [review]

**Review by jamescrowley321 — APPROVED** [review]

**Pull request merged**